src/Core/AbstractApi.php

```diff
@@ -135,4 +135,28 @@ protected function getEndpoint(string $uri, array $query):?string
 
         return $endpoint . (false === \strpos($endpoint, '?') ? '?' : '&') . http_build_query($query);
     }
+
+    final protected function validate($input)
+    {
+        $class = \get_class($input);
+        if (!\defined($class . '::REQUIRED_PARAMETERS')) {
+            // There is nothing we can do
+            return;
+        }
+        $requirements = \constant($class . '::REQUIRED_PARAMETERS');
+        if (!\is_array($requirements)) {
+            throw new InvalidArgument(sprintf('Constant "" must be an array.', $class . '::REQUIRED_PARAMETERS'));
+        }
+
+        foreach ($requirements as $name) {
+            $method = 'get' . $name;
```

Comment:
this could be faster at runtime if we generate a method `isValid()`. 

By implementing the logic here we performs a lot of function call and logic that could have been done when generating the file.

WDYT?

Reply:
I was first planning to have an `AbstractInput` with an `isValid()` function to do these checks. But I changed my mind since I didnt feel like creating a new class =).

Reply:
I'll make sure to generate these.